### ISSUE TITLE
Specific File Will Not Open in 4.2.x, but Will Open in 4.1.x

### ISSUE THREAD

**Issue by Ob-sol** [musescore/MuseScore#21366]
### Issue type

Opening/saving issue

### Bug description

This specific file (File 1) will not open in MS 4.2.x, but it does open in 4.1.x. It is derived from an earlier file I downloaded from musescore.com (File 2) to create the orchestration. The earlier file will open in 4.2.1, but the orchestration will not.

The problem may be with external VSTs, as I used LABS as a sound for the dulcimer, but it's unlikely, as I use it in other projects that work with 4.2.x.

File 1: [Scriabin-Orch copy.mscz.zip](https://github.com/musescore/MuseScore/files/14159189/Scriabin-Orch.copy.mscz.zip)

File 2: [Scriabin_Sonata_No._4_in_F_Major_Movement_I.mscz.zip](https://github.com/musescore/MuseScore/files/14159194/Scriabin_Sonata_No._4_in_F_Major_Movement_I.mscz.zip)

### Steps to reproduce

1. Download File 1
2. Unzip/decompress the file
3. Open the file in Musescore 4.2.x
4. Observe only the palettes, panels, and bars, with no loading of the score even after hours of waiting.

### MuseScore Version

4.2.1

### Regression

Yes, this used to work in a previous version of MuseScore 4.x

### Operating system

macOS Sonoma 14.1.1

### Additional context

The most recent version of Musescore 4.1.x worked well with this file, as I was working on it the day before the update.

**Comment by Jojo-Schmitz** [comment]
It goes into a loop, monopolizing a core (driving it to 100% load)

**Comment by DmitryArefiev** [comment]
Confirm on my side. Both scores (created in 4.1.0 and 4.1.1) don't open in 4.2.0 - 4.3

**Comment by krasko78** [comment]
**SystemLayout::layoutGuitarBends** enters an infinite loop since startOfTie does not change:

_(currently located at line 1406)_
while (startOfTie->tieBack() && startOfTie->tieBack()->startNote()) {
                startOfTie = startOfTie->tieBack()->startNote();
}

**Comment by cbjeukendrup** [comment]
It seems that replacing this code with `Note* startOfTie = note->firstTiedNote();` might help.

**Comment by Jojo-Schmitz** [comment]
https://github.com/musescore/MuseScore/blob/1fe047ceaad04bb14ef7b328a5095b669215b006/src/engraving/rendering/dev/systemlayout.cpp#L1405-L1408
resp.
https://github.com/musescore/MuseScore/blob/1fe047ceaad04bb14ef7b328a5095b669215b006/src/engraving/rendering/stable/systemlayout.cpp#L1369-L1372

**Comment by Jojo-Schmitz** [comment]
```cpp
$ git diff
diff --git a/src/engraving/rendering/dev/systemlayout.cpp b/src/engraving/rendering/dev/systemlayout.cpp
index eb398736ac..5dc7e879b8 100644
--- a/src/engraving/rendering/dev/systemlayout.cpp
+++ b/src/engraving/rendering/dev/systemlayout.cpp
@@ -1402,10 +1402,7 @@ void SystemLayout::layoutGuitarBends(const std::vector<Segment*>& sl, LayoutCont
                 TLayout::layoutGuitarBend(bendBack, ctx);
             }

-            Note* startOfTie = note;
-            while (startOfTie->tieBack() && startOfTie->tieBack()->startNote()) {
-                startOfTie = startOfTie->tieBack()->startNote();
-            }
+            Note* startOfTie = note->firstTiedNote();
             if (startOfTie != note) {
                 GuitarBend* bendBack2 = startOfTie->bendBack();
                 if (bendBack2) {
```
indeed seems to fix it

**Comment by Jojo-Schmitz** [comment]
Let's see whether it passes the tests...

**Comment by DmitryArefiev** [comment]
Fixed in #21574
